src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceServiceTest.kt

```diff
@@ -755,15 +755,33 @@ class LicenceServiceTest {
   }
 
   @Test
-  fun `inactivate licences sets licence statuses to INACTIVE`() {
-    whenever(licenceRepository.findAllById(listOf(1))).thenReturn(listOf(aLicenceEntity.copy(statusCode = LicenceStatus.APPROVED)))
+  fun `activateLicencesByIds calls activateLicences with the licences associated with the given IDs`() {
+    val licence = aLicenceEntity.copy(statusCode = LicenceStatus.APPROVED)
+    whenever(licenceRepository.findAllById(listOf(1))).thenReturn(listOf(licence))
+
+    service.activateLicencesByIds(listOf(1))
+
+    verify(licenceRepository, times(1)).findAllById(listOf(1L))
+    verify(service, times(1)).activateLicences(listOf(licence))
+  }
+
+  @Test
+  fun `activateLicencesByIds filters unapproved licences out before calling activateLicences`() {
+    whenever(licenceRepository.findAllById(listOf(1))).thenReturn(listOf(aLicenceEntity))
+
+    service.activateLicencesByIds(listOf(1))
+
+    verify(licenceRepository, times(1)).findAllById(listOf(1L))
+    verify(service, times(0)).activateLicences(anyList())
```

Comment:
I'm happy to remove the checking for empty lists, but I'd probably still want to verify here that `activateLicences` had been called with an empty list rather than not being called with any list
ie I'd change it to `verify(service, times(1)).activateLicences(emptyList())`

Reply:
yep, that makes sense! I didn't understand that test, 

maybe you make it more explicit by not relying on a default value declared else where?:
```  
    val licence = aLicenceEntity.copy(statusCode = LicenceStatus.UNAPPROVED)
```